xarray/core/dataset.py

```diff
@@ -9204,9 +9209,11 @@ def pad(
             (stat_length,) or int is a shortcut for before = after = statistic
             length for all axes.
             Default is ``None``, to use the entire axis.
-        constant_values : scalar, tuple or mapping of hashable to tuple, default: 0
-            Used in 'constant'.  The values to set the padded values for each
-            axis.
+        constant_values : scalar, tuple, mapping of hashable to tuple or
+            mapping of hashable to mapping of hashable to tuple, default: 0
```

Comment:
This is less consistent but also maybe doesn't require an extreme working memory to understand? What do folks think about mixing the semantic & type info?

```suggestion
        constant_values : scalar, tuple, mapping of var name to tuple or
            mapping of var name to mapping of dim name to tuple, default: 0
```

Reply:
Yeah that's fair, I'll take your suggestion, thank you!

Reply:
with that, we'd have to make `var name` and `dim name` aliases of hashable in `conf.py` to still properly parse and link the type spec

Reply:
I think it is missing `mapping of dim name to tuple` also. So it'd be more like:
 `constant_values : scalar, tuple, mapping of dim name to scalar or tuple, or \
            mapping of var name to scalar, tuple or mapping of dim name to scalar or tuple, default: 0`
Which is still a bit confusing but I think more understandable than before, non?

Reply:
> with that, we'd have to make `var name` and `dim name` aliases of hashable in `conf.py` to still properly parse and link the type spec

I'm not sure if I know how to do this :sweat_smile:

Reply:
> I'm not sure if I know how to do this

Add an entry here: https://github.com/pydata/xarray/blob/6b6a3bd35f134918f29106597ee68a9cc7b31175/doc/conf.py#L127

with the value
```
"dim name": ":term:`dimension name <name>`",
"var name": ":term:`variable name <name>`",
```


Maybe in a new section called "xarray terms"?

Reply:
> > I'm not sure if I know how to do this
> 
> Add an entry here:
> 
> https://github.com/pydata/xarray/blob/6b6a3bd35f134918f29106597ee68a9cc7b31175/doc/conf.py#L127
> 
> with the value
> 
> ```
> "dim name": ":term:`dimension name <name>`",
> "var name": ":term:`variable name <name>`",
> ```
> 
> Maybe in a new section called "xarray terms"?

Ah great thanks for the explanation, will add! :D